UnityProject/Assets/Scripts/Items/Others/Horn.cs

```diff
@@ -0,0 +1,118 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Networking;
+
+/// <summary>
+///     Indicates an object that emits sound upon activation (bike horn/air horn...)
+/// </summary>
+public class Horn : Interactable<HandActivate, PositionalHandApply>
+{
+	public string Sound;
+	public float Cooldown = 0.2f;
+
+	//todo: emit HONK particles
+
+	/// <summary>
+	/// Chance of causing minor ear injury when honking next to a living being
+	/// </summary>
+	[Range(0f, 1f)]
+	public float CritChance = 0.1f;
+
+	[Range( 0f, 100f )]
+	public float CritDamage = 5f;
+
+	private bool allowUse = true;
+	private float randomPitch => Random.Range( 0.7f, 1.2f );
+
+	private IEnumerator CoolDown()
+	{
+		allowUse = false;
+		yield return WaitFor.Seconds(Cooldown);
+		allowUse = true;
+	}
+
+	private void TryHonk(PositionalHandApply clickData = null)
+	{
+		if ( !allowUse )
+		{
+			return;
+		}
+
+		// simple honk on activate
+		if ( clickData == null )
+		{
+			ClassicHonk();
+		}
+		else
+		{
+			bool inCritRange = PlayerScript.IsInReach( clickData.TargetVector );
+			var targetHealth = clickData.TargetObject.GetComponent<LivingHealthBehaviour>();
```

Comment:
it would be better to calculate isCrit first before these two lines

Reply:
it’s not used outside, but I’ll try to make this part easier to understand